book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,266 @@
+might want to consider some changes to make these benchmarks easier and probably
+cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be a specific case or allow more
+flexibility by introducing parameters like for example different circuit sizes.
+We already have most of the infrastructure for this with
+[criterion](https://crates.io/crates/criterion). We would ideally have a
+dedicated computer. However, considering that there aren't that many PRs in
+proof-systems, the requirements shouldn't be that high.
+
+### Kimchi time flamegraph
+
+This can be considered a variant of the [previous](#kimchi-time). Some more
+research is required but this information ideally can be generated from the
+already existing benchmarks with little to no changes.  This may not be as
+interesting to have in CI, but can be useful to run locally for those working in
+optimizations.
+
+### Kimchi instructions
+
+This is basically counting the CPU instructions that takes to run some code,
+could be use as an alternative for CI given that it won't be affected by noise
+and can work in any computer. It would also be faster by running the code only
+once. The implementation will require a bit more research but one option is the
+crate [iai](https://crates.io/crates/iai), with some thinking it could be
+possible to have this and the time benchmarks share the same code.  Is worth
+mentioning that the result are specific to the code and even the computer
+running the benchmark and thus not useful for comparisons with other projects.
+
+### Kimchi memory
+
+Similar to time, but with memory, the constraints are different in some ways,
+noise is not generally a problem and any computer can be used, and also requires
+less sampling and should run faster. This is to some degree already implemented
+through a custom measurement for criterion, and in practice can be slower than
+measuring time, the instrumentation makes it run slower and the some limitations
+in criterion require to run the code more than necessary. With that in mind, the
+best would be to run just a subset of these benchmark for regressions and let
+the others be run at discretion.
+
+### Kimchi memory flamegraph
+
+Similar to the time flamegraph, ideally can be made from the memory benchmark
+without much additional work.
+
+### Snarky time
+
+This is similar to kimchi time, but the fact that it will be in the mina repo
+should be considered, CI will run much more often and costs may be higher, this
+would be a reason to prefer to have benchmarks in kimchi when possible.
+
+### Snarky memory
+
+Here the main interest may be regressions, everything else can probably done at
+the kimchi level more precisely.
+
+### Pickles time
+
+This is similar to snarky time, mostly differing in the overhead of the
+recursion. At this level beyond some regression we will probably have mostly
+application circuits being benchmarked.
```

Comment:
Which applications and how will they be benchmarked specifically?

Reply:
I don't this section should that specific, for tools I decided to use `Core_bench` for ocaml time benchmarks, I can add that. About specific applications that would be in `## benchmarks`. This section is more about a common infrastructure/approach that can be reused for many benchmarks.